### PR TITLE
Few minor improvements for Waterfox Current

### PR DESCRIPTION
**Pull request by hawkeye116477**
Langpacks are from `http://ftp.mozilla.org/pub/firefox/releases/68.1.0esr/linux-x86_64/xpi/` and little modified by me (changed Firefox to Waterfox, etc => https://github.com/hawkeye116477/LanguagePacksForWaterfox/commit/957efac0c4059e1254f2a40f950cbaa4321c03f3). For langpacks we may need maybe some makefile to put them in correct dir, I tried to add them in `browser/features`, but if they are on that dir, seems not working, however they work in `browser/extensions`, but are disabled and can't be enabled until I switch `extensions.autoDisableScopes` to `3`.

I also added Unity/Global Menu support made by Canonical/Ubuntu Mozilla Team (https://bazaar.launchpad.net/~mozillateam/firefox/firefox.xenial/view/1347/debian/patches/unity-menubar.patch) and path to the running binary to `about:support`.

### CODE REVIEW

**Comment by grahamperrin** [comment]
Is the intention here to bundle (include) language packs?

If https://github.com/MrAlex94/Waterfox/issues/1055#issuecomment-543913165 bugging about:preferences can be fixed, then I reckon it'll be better/simpler to have language packs on demand. 

(Recent releases of Firefox take a very smart, effective approach. Quite unlike things as they were a few years ago.)

**Comment by hawkeye116477** [comment]
@grahamperrin Maybe, just like for Classic.
`If #1055 (comment) bugging about:preferences can be fixed, then I reckon it'll be better/simpler to have language packs on demand.`
Putting link to AMO might be not the best thing, cuz some strings needs to be changed for Waterfox.
Better probably will be to put them on some own addons store (`Addons.Waterfox.Net`)/Waterfox's site. For Classic similar thing couldn't be done with provided update url on manifest? As I remember that was some time ago and then it came bundled.

**Comment by hawkeye116477** [comment]
Small update:
Similarly as for Classic  (https://github.com/MrAlex94/Waterfox/pull/1203#issuecomment-544931830), but prefs are on other place. I also hidden default not translated „Restart (developer)" menu item and replaced it with new which also has optional restart confirmation dialog. But maybe restart menu icon needs some improvement.
Screenshot_20191023_171636

**Comment by grahamperrin** [comment]
Are the enhancements to the preferences GUI apparent with `browser.preferences.useOldOrganization` `true`?

Oops, sorry. Current, not Classic.

**Comment by laniakea64** [comment]
Hi @hawkeye116477 , do these new `browser.restart.*` prefs apply to all restarts of Waterfox, or only when using the restart menu items added by this PR?  If the latter, would it be possible to rename the prefs (maybe to `browser.restart_menu.*`) and clarify the `about:preferences` labels to avoid confusion?

**Comment by hawkeye116477** [comment]
@laniakea64 That prefs applies only to restart menu items, similarly as it was on Cyberfox.

`would it be possible to rename the prefs (maybe to browser.restart_menu.*) and clarify the about:preferences labels to avoid confusion?`
Ok, I'll change that later.

**Comment by Peacock365** [comment]
Hey hawkeye116477,

would it be possible to also reinstate the Pop-ups section Cyberfox had? I mean the pop-up, graphics, and javascript settings you can see here:

https://i.computer-bild.de/imgs/4/9/0/7/8/6/2/Screenshot-7-Cyberfox-642x576-5a6b663220ecccae.jpg

After all, the about:config settings _dom.popup_allowed_events_ (for pop-up windows - **EDIT: unnecessary to restore this one, still exists under about:preferences#privacy, that leaves just graphics and Javascript**), _javascript.enabled_ (for javascript), and _permissions.default.image_ (for graphics) still exist, so it shouldn't be hard to bring those settings back.

You would be an absolute hero if something like that were doable for you.

**Comment by hawkeye116477** [comment]
@laniakea64 Changed
Screenshot_20191025_185139


@Peacock365 I'l look into that later.

**Comment by MrAlex94** [comment]
Right, so Mozilla have been testing bundling different languages at build time the same way they do the default locale. I’m going to figure out how to do that because the language packs inflate the binaries size by huge amounts when pre compressed. 

https://firefox-source-docs.mozilla.org/build/buildsystem/locales.html

**Comment by hawkeye116477** [comment]
@MrAlex Or maybe just let users download separate browser and language and set update_url for langpack, so it could be updated automatically?

**Comment by hawkeye116477** [comment]
@MrAlex94 Isn't better to just put langpacks separately from application on some addon store, just like Pale Moon team did?

Anyway, you can accept this PR, all should be good :smile:.

@Peacock365 
Screenshot_20191029_231541
Screenshot_20191029_231603

**Comment by Peacock365** [comment]
@hawkeye116477 Fantastic work!

**Comment by hawkeye116477** [comment]
Small preview what I done:
Screenshot_20191103_190456
Screenshot_20191104_153750


If I want to also move some settings to another categories, make more categories, mainly move from general to another categories, little like on Vivaldi, should I do that in current preferences or make „new order" toggleable just like on Classic?

**Comment by grahamperrin** [comment]
> move some settings to another categories, 

-1

> make more categories, 

+1 

– e.g. fields for strings to make location work

> mainly move from general to another categories, 

-1

> should I do that in current preferences 

-1

> or make „new order" toggleable just like on Classic?

+1

– and default to normal (Firefox-like) order. 

HTH

**Comment by hawkeye116477** [comment]
Ok. I wanted to move settings related to Tabs to new big category „Tabs", create also „Appearance" category (maybe in the future would come here some features from CTR if plan is to integrate it) , „Files and applications", maybe „Webpages", all-new settings „Sidebar" and „Display all settings".

And question is now, so for Classic should I did third preferences pane, so will be „old", „new" and „new order"? That seems rather not that good, I mean more code to maintain, but seems that no volunteers for maintaining „new".

-----
I added now possibility to move tab bar, for now there is „Top above address bar", „Top under address bar" and „Bottom", maybe in the future will be also left and right position, but seems that's more fun with css, so I didn't added that for now.

**Comment by hawkeye116477** [comment]
All-new preferences pane. I set it as not default.
Screenshot_20191109_150617

**Comment by grahamperrin** [comment]
>  Remove annoying message about private browsing on addons page 

From the commit, I can't visualise the message. Can you provide a screenshot? 

Thanks.

**Comment by hawkeye116477** [comment]
@grahamperrin Set `extensions.htmlaboutaddons.enabled` to false and you'll see that message.

**Comment by grahamperrin** [comment]
The purple indicators? 





It's important to **not** lose those completely.

**Comment by hawkeye116477** [comment]
@grahamperrin Not purple, I mean that message „Waterfox Current is changing...".

**Comment by grahamperrin** [comment]
If that's removed, then what's left to educate the user? 

In Firefox 70.0.1, a click on a purple icon of privacy refers to e.g.:

https://support.mozilla.org/1/firefox/70.0.1/FreeBSD/en-GB/extensions-pb

– which redirects to: 

https://support.mozilla.org/en-US/kb/extensions-private-browsing?as=u&utm_source=inproduct

**Comment by hawkeye116477** [comment]
@grahamperrin @Peacock365 requested, so I removed, but users will manage :smile:
There is also inactive link to add-ons supports.
Maybe I'l look to that later and restore that links, which currently refer to nothing.

**Comment by Peacock365** [comment]
Uhm, @grahamperrin, this silly notification takes up a whole lot of screen real estate. This is unacceptable in about:addons, which is meant to be a rather compact overview of your add-ons. Also, the new HTML about:addons page doesn't display this notification, either. It's also unacceptable to have that notification there for the entirety of the next year. When you click on an add-on, you can choose whether to enable it in private browsing windows or not, the switch is rather self-explanatory as well.

**Review by MrAlex94 — COMMENTED** [review]
I'm weary about adding this change - which is why I didn't do it in the first place. brandShortName is used quite often [throughout](https://searchfox.org/mozilla-esr68/search?q=brandShortName&path=) the codebase, and I don't know how this might break users browser space. It has happened before where branding changes have messed things up, so this will need extensive testing.

**Review by MrAlex94 — DISMISSED** [review]

**Comment by hawkeye116477** [comment]
@MrAlex94 That must be added => https://github.com/MrAlex94/Waterfox/pull/1213/commits/46ac9442f23453ccb53f75c49ed5a8d038799349, cuz without it, on Linux there is icon mish mash (https://github.com/MrAlex94/Waterfox/issues/1211).

**Comment by hawkeye116477** [comment]
@MrAlex94 So should I revert https://github.com/MrAlex94/Waterfox/pull/1213/commits/46ac9442f23453ccb53f75c49ed5a8d038799349? Anyway without that change - on Linux is icon mish mash  (https://github.com/MrAlex94/Waterfox/issues/1211), there is also not distinctive title on Profile Manager (https://github.com/MrAlex94/Waterfox/issues/1247).

`this will need extensive testing.`
Isn't that `the next generation of Waterfox for testing`?

Is there something another except this change which you didn't approve?

**Comment by MrAlex94** [comment]
> Is there something another except this change which you didn't approve?

Not yet, but I need to test each change thoroughly. It's taking a while because there are a lot of changes here.

**Comment by hawkeye116477** [comment]
I'd say not that a lot, only a lot code. Anyway, ok.

**Comment by MrAlex94** [comment]
Well, it becomes a lot when testing each change on every platform :-) plus
I’ve been busy with the website so it isn’t going as fast as it could.
Appreciate the work though, won’t be long now.

File: browser/components/preferences/in-content-all-new/webpages.js [hunk 1]
```diff
@@ -0,0 +1,1930 @@
...
+  updateHardwareAcceleration() {
+    // Placeholder for restart on change
+  },
+
+  destroy() {
+    window.removeEventListener("unload", this);
+    Services.prefs.removeObserver(PREF_SHOW_PLUGINS_IN_LIST, this);
+    Services.prefs.removeObserver(PREF_HIDE_PLUGINS_WITHOUT_EXTENSIONS, this);
+
+    Services.obs.removeObserver(this, AUTO_UPDATE_CHANGED_TOPIC);
```
Comment (MrAlex94) [inline browser/components/preferences/in-content-all-new/webpages.js]: This returns a `NS_ERROR_ILLEGAL_VALUE`. Is that expected?

Comment (hawkeye116477) [reply browser/components/preferences/in-content-all-new/webpages.js]: No, my bad, fixed now.

**Comment by grahamperrin** [comment]
> … this silly notification takes up a whole lot of screen real estate …

Agreed re: the use of space, which is why I drew attention to the sensible Firefox behaviour in response to clicks on the small purple icons.

**Comment by MrAlex94** [comment]
Okay, have been testing and all seems well. Will pull in a bit.

File: browser/components/customizableui/content/panelUI.js [hunk 2]
```diff
@@ -276,6 +276,15 @@ const PanelUI = {
...
       case "popupshowing":
         updateEditUIVisibility();
+        try {
+          if (!Services.prefs.getBoolPref("browser.restart_menu.showpanelmenubtn")) {
+            document.getElementById("appMenu-restart-button").hidden = true;
+          } else {
+            document.getElementById("appMenu-restart-button").hidden = false;
+          }
+        } catch (e) {
+          throw new Error("We're sorry but something has gone wrong with 'browser.restart_menu.showpanelmenubtn'" + e);
```
Comment (MrAlex94) [inline browser/components/customizableui/content/panelUI.js:286]: This error keeps getting thrown when accessing the hamburger menu on macOS. Tested with both restart button enabled/disabled.

`JavaScript error: chrome://browser/content/customizableui/panelUI.js, line 286: Error: We're sorry but something has gone wrong with 'browser.restart_menu.showpanelmenubtn'TypeError: document.getElementById(...) is null`

Comment (hawkeye116477) [reply browser/components/customizableui/content/panelUI.js:286]: Ah, I see the problem, I added `appMenu-restart-button` inside of condition `#ifndef XP_MACOSX`, so it's not available for Mac.

Comment (MrAlex94) [reply browser/components/customizableui/content/panelUI.js:286]: Cool, thanks for the fix.

**Comment by MrAlex94** [comment]
Since I used most of the commits here (but not all), I `git cherry-pick`'ed the commits and then pushed them to the `current` branch. It doesn't look like GitHub knows how to handle partial commit pulls.

**Comment by Peacock365** [comment]
@hawkeye116477 @MrAlex94 

I've noticed that, when one of the australized themes (curved tabs) is used, the new tab button in tab bar is squared, I suppose this is not intended to be so, since the new tab button behind the tabs was rounded in Waterfox 68b1... Please fix this cosmetic issue.

**Comment by hawkeye116477** [comment]
@MrAlex94 In case of https://github.com/MrAlex94/Waterfox/commit/2ba3e023e1a8c50610e45bafe7c380609ef75246, if you wanted FF-like behaviour, then `browser.statusbar.mode` should be set to `1`, cuz `0` disables showing any status.

**Comment by Peacock365** [comment]
Kind reminder that the "new tab button cosmetic issue" still is not fixed.

**Pull request closed without merging**